### ISSUE TITLE
[Feature Request] Vis Interface

### ISSUE THREAD

**Issue by Silly511** [Nividica/ThaumicEnergistics#232]
I think you should add Vis Interfaces. Vis Interfaces would be able to to two things:

1. You could place them on a node and then apply a redstone signal and the vis interface would drain the node of all of it's vis and store it in the network as essentia.

2. You could place a vis relay on the is interface and any thing connected to the vis relay would have access to all the essentia/vis stored in the network.

Also the Vis Interface would be really hard to craft since it can convert essentia to vis.

**Comment by XFactHD** [comment]
Coverting vis into essentia and backwards would break the balance of thaumcraft (vis is raw magical energy and essentia is magical essence in form of a gas).

If you have an energized node and use a vis relay interface with a vis relay on top, it can drain the energy out of the energized node and you can connect the vis relay interface to the arcane assembler (replaces the need of a charged wand), the arcane crafting terminal (charges the wand in the terminal) and also to other vis relay interfaces (to power things like the focus manipulator over long distances).

**Comment by Silly511** [comment]
I know to would mess up the balance of thaumcraft. But I thought that it could be a really late game thing. Because all you need is a hunger node and a vis interface and infinite vis/essentia.

**Comment by XFactHD** [comment]
You do not need a hungry node, you can make an energized node out of any normal node, the only benefit of the hungry node is that you can feed it to grow. "Really late game thing" and "really hard to craft" are no arguments anymore in the modpacks nowadays. Micro crafting would just be stupid and the ressource cost also doesn't play a role anymore.

**Comment by Nividica** [comment]
As XFactHD mentioned, essentia and vis disparate things. Think of it as matter and energy. Now while out here IRL we have the famous Einstein equation that describes the relationship between these two, no such relationship has been found in the arcane. If however there was such a relationship, creating vis would be a much simpler process than creating essentia as a raw conversion.
The creation of vis could be realized with a reactor of-sorts, releasing the cohesion of matter into primal energetic vis forms. The immense amount of released vis would make any hope of storing more than ~1-5% of it unfeasible but not necessarily impossible. Another point to consider is that the infrastructure required to maintain a reactor of this caliber would be vast. One would likely need to take advantage of the trans-dimensional nature of vis nodes and construct a containment system that uses the released vis as the means of containing itself, rather than a static container. (lookup diamagnetic for a better idea of what I am talking about)
Assembling a single unit of essentia from vis would require much, much more work. Not only would it require an unimaginable amount of vis but keeping the energy flows and internal lay lines from twisting and kinking would require spatial manipulation of the highest precision.
The simplest comparison between the two would be to look at vis creation like energy creation from a bomb. It is simple to create(not harness) a large amount of energy from setting off the bomb, however in contrast it is by no means simple to take the released energy and reassemble the exploded bomb.
For now your best bet of gathering essentia is to take advantage of the existing materials in the world and rely on simpler reactions such chemical bonding, rather than attempting to control matter-energy cohesion.

TLDR: Essentia->Vis may be coming in the future via the power reactor, but not Vis->Essentia.
